### PR TITLE
extend FileInput with more attributes.

### LINKED ISSUE #9099: [FEATURE] Extend FileInput to return filename and make output clearer
I have been playing with the new FileInput widget in the 1.3.0rc2 release and have 2 suggestions/needs I found I had:

- Return the filename of the file being uploaded, since I need to display the names of imported files back to the users.
- Return the actual base64 encoded string of the file content separately from the mime type, rather than combined as they are at the moment. (something like this `data:application/json;base64,ewogICAgInBlcnNvbiI6IHsKICAgICAgICAiYWdlIjogMjIKICAgIH0KfQo=`)

I have a working version that does both of these so if its wanted I can create a pull request to maybe get into the 1.3.0 main release.

#### Background:
Following #9083 / #6096 I saw that the new widget is based on the Panels one, but it seems they have some sort of logic to split up the file string in the python part of the code after the ts bit. The new bokehjs code just does this logic in the ts instead.

### PR DESCRIPTION
**Pull request by jbeanland**
Replaces the `file` attribute in the FileInput widget with separate `value` and `mime_type` attributes. Also adds a `filename` attribute.

- [x] issues: fixes #9099
- [ ] tests added / passed
- [ ] release document entry (if new feature or API change)

### CODE REVIEW

**Comment by bryevdv** [comment]
This looks good, if you have a little more bandwidth a simple example in `examples/plotting/file` would cap this off. No worries if not, though.

**Comment by jbeanland** [comment]
Thanks, I should be able to manage that, I'll fix these failed checks too, looks like my introduction to Typescript didn't go perfectly.

File: bokehjs/src/lib/models/widgets/file_input.ts [hunk 1]
```diff
@@ -28,19 +28,30 @@ export class FileInputView extends WidgetView {
 
   load_file(e: any): void {
     const reader = new FileReader()
+    this.model.filename = e.target.files[0].name
     reader.onload = (e) => this.file(e)
     reader.readAsDataURL(e.target.files[0])
   }
 
   file(e: any): void {
```
Comment (mattpap) [inline bokehjs/src/lib/models/widgets/file_input.ts]: If you are at it, you could also improve those `any` types. This one should be replaced with `ProgressEvent`. You can learn precise types from https://github.com/microsoft/TypeScript/blob/master/lib/lib.dom.d.ts (interface `FileReader`).

**Comment by jbeanland** [comment]
I've started hitting walls. There's a filing integration test, with tests/codebase/test_code_quality.py failing for a trailing whitespace check but locally that one passes.

On the example front I hit a weird thing, maybe some sort of bug in CustomJS or maybe I just missed some sort of build step:

If I try to setup a CustomJS callback on a file I'm outputting with `show()` then the callback itself acts as if the new attributes don't exist but the old one (`file`) does. The widget itself is correct because if I try to initiate it with something for `file` it will fail on AttributeError.

But if I run the same thing using `bokeh serve` then the callback will work properly.

**Comment by bryevdv** [comment]
@jbeanland are you setting `BOKEH_RESOURCES=inline`? By default Bokeh uses BokehJS from CDN for standalone content, so if you want to use a locally built BokehJS you have to tell it to

**Comment by bryevdv** [comment]
@jbeanland For the linter error, merge/rebase on master I was editing the README in the web UI and left a trailing space on one commit evidently (but its fixed now)

**Comment by bryevdv** [comment]
@jbeanland one last thing the new example needs a baseline. Can you please check in an empty file at

> tests/baselines/examples/plotting/file/file_input

Then after the baseline tests fails again, it will print a diff and the file can be updated with the actual proper value

**Comment by jbeanland** [comment]
@bryevdv Beauty, I've just been trying to figure out how to make those work. I'll check that in and hopefully(...) we'll be there.

**Comment by jbeanland** [comment]
@bryevdv so to confirm I guess, I should wait for the inevitable failure, go get the diff, put that diff in that new empty file, then recommit with the new, expected test data in there?

**Comment by bryevdv** [comment]
@jbeanland that is 100% correct!

**Comment by bryevdv** [comment]
I paused a few other jobs to get this one up and running sooner: https://travis-ci.org/bokeh/bokeh/jobs/561216250

**Comment by bryevdv** [comment]
@jbeanland The diff from the faillure is here:

https://travis-ci.org/bokeh/bokeh/jobs/561216250#L2884

**Comment by jbeanland** [comment]
Awesome! Done.

**Comment by bryevdv** [comment]
Thanks @jbeanland !

**Comment by jbeanland** [comment]
@bryevdv No worries, happy to be able to help with the project! Thanks for guiding me through it.

**Pull request merged**